### ISSUE TITLE
Bird hostname has changed, documentation outdated

### ISSUE THREAD

**Issue by mxdr** [ubahnverleih/WoBike#175]
https://github.com/ubahnverleih/WoBike/blob/master/Bird.md

Bird has changed the hostname for listing scooters. Luckily it's a simple fix.

OLD: `api.birdapp.com`
NEW: `api.prod.birdapp.com`

The paths seem to be the same, though.

**Comment by watzon** [comment]
Anyone else finding it impossible to connect even with api.prod.birdapp.com? Things were working fine 4 days ago, then I started getting 403 errors from cloudfront.

**Comment by mxdr** [comment]
@watzon I have wrestled a lot with my integration with bird, usually it's that they have banned either your IP or your token because you are calling the API too frequently. So if I were you, I'd try to generate a new token (WoBike documentation works in that regard). If that doesn't work, try to make a request with a different IP. Proxies have been helpful for me.

Also make sure you allow sufficient time between calls, I recommend 1 second but it may work with as little as 200ms between calls.

I read through my bird integration, which works fine. And it doesn't seem to differ from the WoBike documentation besides the hostname of the `/bird/nearby` calls.

**Comment by watzon** [comment]
Hmm, I’ve tried just about everything I can think of, down to using a rotating proxy. It seems like I get the 403 regardless of whether it’s my IP or a proxy IP. I’ll keep trying though, thanks!

**Comment by mxdr** [comment]
@watzon make sure any proxy you test is fully anonymous. Otherwise your real IP will leak and bird (cloudfront?) will detect it.

Are you only getting 403s for the `/bird/nearby` calls or are you getting a 403 from the refresh endpoint?

**Comment by watzon** [comment]
All api-auth endpoints work fine, it’s just api.prod.bird.co endpoints that aren’t working. As far as those go, it doesn’t seem like any are working. nearby, zones, bounties, etc. all result in a 403.

**Comment by mxdr** [comment]
Strange. Have you tried changing your Device-Id?

These are my headers for those calls (removed my Device-Id and Token) if you wanna double check:

```
{
   "Content-Type":"application/json",
   "User-Agent":"Bird/4.53.0 (co.bird.Ride; build:24; iOS 12.4.1) Alamofire/4.53.0",
   "Device-Id":"YOUR_GUID",
   "Platform":"ios",
   "App-Version":"4.53.0",
   "Authorization":"Bird YOUR_TOKEN,
   "Location":"{\"latitude\": 12.345678, \"longitude\": 23.456789, \"altitude\": 0, \"accuracy\": 100, \"speed\": -1, \"heading\": -1}"
}
```

and this is the urI: `https://api.prod.birdapp.com/bird/nearby?latitude=12.345678&longitude=23.456789&radius=1000`

**Comment by watzon** [comment]
Yeah I've changed it a few times. I'll dig into it deeper tonight and see if I can figure it out. The fact that it's still working for you is good news.

**Comment by watzon** [comment]
Yeah this is weird. I'm hitting this URL: `https://api.prod.birdapp.com/bird/nearby?latitude=40.7675818&longitude=-111.8889092&radius=100` with these headers:

```json
{
  "Location": "{\"latitude\":40.7675818,\"longitude\":-111.8889092,\"altitude\":0,\"accuracy\":100,\"speed\":-1,\"heading\":-1}",
  "User-Agent": "Bird/4.53.0 (co.bird.Ride; build:24; iOS 12.4.1) Alamofire/4.53.0",
  "Device-Id": "59de9bd4-74ca-4b54-bc4b-2039f5c98363",
  "Platform": "ios",
  "App-Version": "4.53.0",
  "Content-Type": "application/json",
  "Authorization": "Bird MY_TOKEN"
}
```

The token is definitely accurate (just generated it and validated that it's a proper JWT) and I don't think it's an IP issue (I tried from a different network in a different city tonight.

**Comment by watzon** [comment]
I did end up figuring out my issue. I had everything right, but in my client I was also providing an empty JSON string as a body to the GET request. Before CloudFront was handling this fine, but apparently now it's not. Pretty easy fix, but really hard to debug.

**Comment by KurtWoloch** [comment]
The nasty thing is... the old host still works (for me), but the data it spits out is complete garbage... the scooters in the output change every time you request data for the same location, and the coordinates lay on a straight line going to the northeast from the specified point.

**Comment by ricffb** [comment]
@KurtWoloch I had the same problem. I managed to fix it in #195 . 
I played around with the headers and scanned their connections again (with charles). I could pin the weird scooters down to
a missing user-agent header.
